include/rocksdb/iterator.h

```diff
@@ -117,6 +117,42 @@ class Iterator : public Cleanable {
     assert(false);
     return Slice();
   }
+
+  /**
+   * true if this iterator is already
+   * checking ReadOptions::iterate_lower_bound,
+   * false otherwise.
+   */
+  virtual bool check_lower_bound() const { return false; }
+
+  /**
+   * Returns the lower bound if this iterator has a
+   * ReadOptions::iterate_lower_bound set, else
+   * returns nullptr.
+   *
+   * Just because a lower bound is present, it does
+   * not mean that it is checked. This can however
+   * be determined by calling Iterator::check_lower_bound().
+   */
+  virtual const Slice* iterate_lower_bound() const { return nullptr; }
```

Comment:
Also a verb, where the reader is supposed to understand from the use of lowercase_with_underscores that this is an accessor not an action. Since we're already in an iterator, can we just call it `lower_bound`?